### DISCUSSION TITLE
Alternative to add_linear_dim

### DISCUSSION THREAD

**Discussion by jrflack** [mozman/ezdxf#579] [Q&A]
Hi, I have a code structure like a tree. I want each "leaf" to draw itself and add dimension information.

Currently EZDXF is structured so annotations can be added at the msp object, i.e. msp.add_linear_dim(parameters...).

Is there a way I can create "remote" linear dim object, then parse it to the msp to add later?

Example:
Leaf1.linear_dim = EZDXF.linear_dim_object(p1, p2, angle, etc...).

Leaf1.linear_dim = EZDXF.linear_dim_object(p1, p2, angle, etc...).

Msp.add(Leaf1.linear_dim, Leaf2.linear_dim)

Thanks Joseph. 

I'm a mechanical engineer who knows the valve of automation! Anything a keyboard and mouse can do can be automated :)

**Comment by mozman** [comment]
The problem is the complexity of the DIMENSION entity. It is not just an entity it also requires an graphical representation of DXF primitives (rendering) in an anonymous BLOCK. You can study how this is done in the `graphicsfactory.py` module and replicate this behavior.

But I would suggest to collect the required parameters an add them later by the "official" factory methods. For more information look at the linear dimension [tutorial](https://ezdxf.mozman.at/docs/tutorials/linear_dimension.html).

**Comment by jrflack** [comment]
Thanks!
